docs/src/vjp_laws.jl

```diff
@@ -0,0 +1,377 @@
+# # Law VJP customization
+
+# This tutorial explains how to customize VJP (vector-Jacobian product) computation of the laws in ODINN.jl and clarifies the runtime flow used internally by the library.
+# It explains which functions are part of the public, user-facing customization API and which are internal helpers used by ODINN when an automatic-differentiation (AD) backend is required.
+
+# It assumes that you have followed the [Laws](./laws.md) tutorial.
+
+using ODINN
+rgi_ids = ["RGI60-11.03638"]
+rgi_paths = get_rgi_paths()
+params = Parameters(
+    simulation = SimulationParameters(rgi_paths=rgi_paths),
+    UDE = UDEparameters(grad=ContinuousAdjoint()),
+)
+# params = Parameters()
+nn_model = NeuralNetwork(params)
+
+# ## Explanations
+
+# ### High-level summary
+
+# At the user level the customization can be made by implementing hand-written VJPs through the following functions:
+# - `f_VJP_input!(...)` — VJP w.r.t. inputs
+# - `f_VJP_θ!(...)` — VJP w.r.t. parameters θ
+# - You may also implement your own precompute function to cache expensive computations which is the purpose of `p_VJP!(...)`. This function is called before solving the adjoint iceflow PDE.
+
+# Internally when the user does NOT provide VJPs, ODINN uses a default AD backend (via [DifferentiationInterface.jl](https://github.com/JuliaDiff/DifferentiationInterface.jl)).
```

Comment:
Just to be clear, does this work already with DifferentiationInterface? I understood that it would take a while to adapt all the interfaces.